### PR TITLE
finish db test + add web test

### PR DESCRIPTION
**Pull request by sidpremkumar**
## Description of the change

Untested, but adds web system tests. And finishes the db connect portion of the db system test. 

## Relevant release note information

Release Notes:

## Related JIRA tickets

Relates to JIRA: CWC-XXX

## Have you considered the security impacts?

Does this PR have any security impact?

- [ ] Yes
- [ ] No

If yes, please explain:

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -88,11 +87,13 @@
     "open": "^7.3.0",
     "openid-client": "^4.7.2",
     "ora": "5.4.1",
+    "pgtools": "^0.3.2",
```
Comment (lipmas) [inline package.json:90]: We can make this a devDependency

File: src/system-tests/tests/suites/vt.ts [hunk 2]
```diff
@@ -56,7 +53,52 @@ export const vtSuite = () => {
...
+            // Start the connection to the db virtual target
+            await callZli(['connect', webVtName]);
+
+            const testConnectionRequest = await got.get(`http://localhost:${localWebPort}/`, { throwHttpErrors: false, https: { rejectUnauthorized: false } });
+
+            if (testConnectionRequest.statusCode === 200) {
+                return;
+            } else {
+                throw new Error(`Bad status code when testing web VT connection: ${testConnectionRequest.statusCode}`);
+            }
```
Comment (lipmas) [inline src/system-tests/tests/suites/vt.ts:101]: you can simply make this a single jest expect assertion like:
```
expect(testConnectionRequest.statusCode).toBe(200);
```

File: src/system-tests/tests/suites/vt.ts [hunk 3]
```diff
@@ -56,7 +53,52 @@ export const vtSuite = () => {
...
 
-            // TODO: verify psql to localhost:6100 can connect to the database
+            const credentials = {
+                user: "postgres",
+                host: "localhost",
+                password: "",
+                port: localDbPort,
+            };
+
+            pgtools.createdb(credentials, "some_db", function(err: any, res: any) {
```
Comment (sidpremkumar) [inline src/system-tests/tests/suites/vt.ts:63]: Is this going to have to turn into a promise that we `resolve` or `reject`?

File: src/system-tests/tests/suites/vt.ts [hunk 4]
```diff
@@ -56,7 +53,52 @@ export const vtSuite = () => {
...
+                if (err) {
+                  throw err
+                }
+                return;
+              });
```
Comment (lipmas) [inline src/system-tests/tests/suites/vt.ts:68]: Looks like pgtools supports promise API too so you i think we could just write this as:

```
await pgtools.createdb(credentials, "some_db");
```

Comment (sidpremkumar) [reply src/system-tests/tests/suites/vt.ts]: https://github.com/bastionzero/zli/pull/303/commits/11bbaabeb95e77f94c5c32682d682673d44687b2

File: src/system-tests/tests/suites/vt.ts (hunk 2 above)
Comment (sidpremkumar) [reply src/system-tests/tests/suites/vt.ts]: https://github.com/bastionzero/zli/pull/303/commits/11bbaabeb95e77f94c5c32682d682673d44687b2

File: package.json (hunk 1 above)
Comment (sidpremkumar) [reply package.json]: https://github.com/bastionzero/zli/pull/303/commits/11bbaabeb95e77f94c5c32682d682673d44687b2

**Review by lipmas — APPROVED** [review]

**Pull request merged**